### ISSUE TITLE
Replace 'Choose a displayname' dialogue with a dialogue to choose a MXID

### ISSUE THREAD

**Issue by lampholder** [vector-im/riot-web#3604]
The details of the new guest experience for Riot are on the project plan: vector-im/riot-meta#59

Instead of prompting unregistered users to pick a display name (for their automatically-generated @guest_123456:matrix.org mxid), instead we will invite users to pick a new mxid (for now throwing away their generated guest ID, which is fine because it won't have been allowed to say anything).

This dialogue should __handle 'username in use' gracefully__, ideally checking username availability in real time.

**Comment by lampholder** [comment]
Some UX suggestions:

Invitation to choose a username:
![set_username](https://cloud.githubusercontent.com/assets/1922197/24921412/ac009b8c-1ee2-11e7-9668-f772e4b19708.png)

Appearance if chosen username is available:
![set_username_available](https://cloud.githubusercontent.com/assets/1922197/24921411/ac004c4a-1ee2-11e7-8713-457d588547ee.png)

Appearance if chosen username is unavailable:
![set_username_unavailable](https://cloud.githubusercontent.com/assets/1922197/24921413/ac00e3e4-1ee2-11e7-96bc-00dc78cd41e3.png)

Outstanding things to consider:
- When we check availability (with every keypress? After _n_ seconds?)
- How we communicate permitted/otherwise characters (on keypress? Does it need to go into prose?)

**Comment by lukebarnard1** [comment]
This will require Riot to be able to generate user passwords: #3600

**Comment by lampholder** [comment]
@AmandineLP @ara4n - what are your thoughts on the proposed wording/UX?

**Comment by ara4n** [comment]
So, i think this was a bit too wordy and scary, how about:

---

### To get started, please pick a username!

     _________________________

This will be your account name on the $foo homeserver, or you can pick a [different server](/#login).

---

After picking a username, we'd obviously do whatever captcha dance is needed (if configured).

* We show this whenever the user is moving from read-only (i.e. 'guest') to read-write mode.  In practice:
  * If they followed permalink of some kind, it's the point that they hit 'join' to actually join the room - i.e. the point we currently say "Please pick a display name" for Guests
  * If they just entered riot.im/app itself, the app will immediately show this prompt (over a 'blank' UI), and having created the account, the app will prod RiotBot to send an invite to the new user, which the app will autoaccept, and so the first screen the user sees will be RiotBot.
  * Alternatively, if the Riot deployment doesn't want to use RiotBot, then they should probably dump the user into RoomDirectory by default or a welcome page (https://github.com/vector-im/riot-web/issues/3806, or the RTS welcome page).

For RTS:
 * The user is dumped into the RTS as a guest (i.e. read-only mode).  The LeftPanel is empty.
 * If they try to join a room, then they get bumped to the full login flow - which prompts them for mxid, an email, auths them to join the team, does captcha.
 * Having logged them in, we return them to the RTS Welcome Page, autojoin the RTS-defined rooms, and prod RiotBot to invite the user.  (Alternatively we could route them to RiotBot; unsure which).

**Comment by lukebarnard1** [comment]
- [ ] Redesign as above
 - [ ] Hit username availability API on Synapse and show ✅ or ❎

**Comment by lukebarnard1** [comment]
I'm not sure what [different server](url) is supposed to do.

**Comment by lampholder** [comment]
In terms of user flow, we decided that users who want to sign in to a different server should go via the existing registration flow (there's no need for the streamlined signup to account for less common usecases, at least to begin with).

That said, it's clear that we're missing a non-jarring way to get them there.

Clicking on [different server](url) might:
- dump you straight on https://riot.im/develop/#/register perhaps with the custom server bit preopened
- expand a hidden paragraph explaining what's going on and how you need to go via the standard registration for a custom homeserver
- link to a blog post/video explaining about custom servers and how you need to go to https://riot.im/develop/#/register to use them

Any other thoughts?

**Comment by lukebarnard1** [comment]
>dump you straight on https://riot.im/develop/#/register 
Is now the current behaviour

**Comment by lukebarnard1** [comment]
What happens once the user has chosen a username? 
 1. Riot retries the same action but with the new user session
 2. Show the RiotBot or if unconfigured, the room directory?
